app/services/katello/pulp3/api/core.rb

```diff
@@ -6,8 +6,9 @@ module Api
       class Core
         attr_accessor :smart_proxy
 
-        def initialize(smart_proxy)
+        def initialize(smart_proxy, repo = nil)
```

Comment:
What do you think about overriding the initializer for core in generic.rb and making the repository argument required?

So then initialize in generic.rb could look like:
```ruby
def initialize(smart_proxy, repository)
  @repository = repository
  super(smart_proxy)
end
```

Reply:
I'm suggesting that since it would be good to not modify core unless we really have to. Plus, it'll make it more apparent that the Generic API needs to have the repo passed in.